dm/pkg/checker/privilege.go

```diff
@@ -14,58 +14,54 @@
 package checker
 
 import (
+	"bytes"
 	"context"
 	"database/sql"
 	"fmt"
-	"strings"
 
-	"github.com/pingcap/errors"
 	"github.com/pingcap/tidb-tools/pkg/dbutil"
 	"github.com/pingcap/tidb/parser"
 	"github.com/pingcap/tidb/parser/ast"
 	"github.com/pingcap/tidb/parser/mysql"
 	_ "github.com/pingcap/tidb/types/parser_driver" // for parser driver
-)
-
-var (
-	dumpPrivileges = map[mysql.PrivilegeType]struct{}{
-		mysql.ReloadPriv: {},
-		mysql.SelectPriv: {},
-	}
-	replicationPrivileges = map[mysql.PrivilegeType]struct{}{
-		mysql.ReplicationClientPriv: {},
-		mysql.ReplicationSlavePriv:  {},
-	}
+	"go.uber.org/zap"
 
-	// some privileges are only effective on global level. in other words, GRANT ALL ON test.* is not enough for them
-	// https://dev.mysql.com/doc/refman/5.7/en/grant.html#grant-global-privileges
-	privNeedGlobal = map[mysql.PrivilegeType]struct{}{
-		mysql.ReloadPriv:            {},
-		mysql.ReplicationClientPriv: {},
-		mysql.ReplicationSlavePriv:  {},
-	}
+	"github.com/pingcap/tiflow/dm/pkg/log"
 )
 
-/*****************************************************/
+// some privileges are only effective on global level. in other words, GRANT ALL ON test.* is not enough for them
+// https://dev.mysql.com/doc/refman/5.7/en/grant.html#grant-global-privileges
+var privNeedGlobal = map[mysql.PrivilegeType]struct{}{
+	mysql.ReloadPriv:            {},
+	mysql.ReplicationClientPriv: {},
+	mysql.ReplicationSlavePriv:  {},
+}
 
 // SourceDumpPrivilegeChecker checks dump privileges of source DB.
 type SourceDumpPrivilegeChecker struct {
-	db     *sql.DB
-	dbinfo *dbutil.DBConfig
+	db          *sql.DB
+	dbinfo      *dbutil.DBConfig
+	checkTables map[string][]string // map schema => {table1, table2, ...}
+	consistency string
 }
 
 // NewSourceDumpPrivilegeChecker returns a RealChecker.
-func NewSourceDumpPrivilegeChecker(db *sql.DB, dbinfo *dbutil.DBConfig) RealChecker {
-	return &SourceDumpPrivilegeChecker{db: db, dbinfo: dbinfo}
+func NewSourceDumpPrivilegeChecker(db *sql.DB, dbinfo *dbutil.DBConfig, checkTables map[string][]string, consistency string) RealChecker {
+	return &SourceDumpPrivilegeChecker{
+		db:          db,
+		dbinfo:      dbinfo,
+		checkTables: checkTables,
+		consistency: consistency,
+	}
 }
 
 // Check implements the RealChecker interface.
-// We only check RELOAD, SELECT privileges.
+// We check RELOAD, SELECT, LOCK TABLES privileges according to consistency.
 func (pc *SourceDumpPrivilegeChecker) Check(ctx context.Context) *Result {
 	result := &Result{
 		Name:  pc.Name(),
 		Desc:  "check dump privileges of source DB",
-		State: StateFailure,
+		State: StateSuccess,
```

Comment:
can we set the default state to failure? currently we can make sure for each error path's `return` , we have marked the state as failure. But that's error-prone when we add more error path in future.